File: mobile/app/src/main/java/com/geneirodan/nextgenmeal/data/Order.kt
package com.geneirodan.nextgenmeal.data

import androidx.annotation.Keep
import kotlinx.datetime.LocalDateTime
import kotlinx.serialization.Serializable

@Serializable
data class Order(
    val id: Int,
    val price: Double,
    val status: String,
    val time: LocalDateTime,
    val isBox: Boolean,
    val customerId: String?,
    val cateringId: Int,
    val orderDishes: List<OrderDish>,
    val catering: Catering
)

File: server/DataAccess/Entities/Order.cs
﻿using DataAccess.Entities.Users;
using Microsoft.EntityFrameworkCore;

namespace DataAccess.Entities
{
    public class Order : Entity
    {
        [Precision(18, 2)]
        public decimal Price { get; set; }
        public string Status { get; set; } = null!;
        public DateTime Time { get; set; }
        public bool IsBox { get; set; }
        public string? CustomerId { get; set; }
        public virtual Customer? Customer { get; set; }

        public int? CateringId { get; set; }
        public virtual Catering? Catering { get; set; }

        public virtual List<OrderDish> OrderDishes { get; set; } = null!;

        public override string? GetOwnerId() => CustomerId ?? Catering?.GetOwnerId();
    }
}
